provekit/prover/src/whir_r1cs.rs

```diff
@@ -245,20 +260,34 @@ pub fn run_zk_sumcheck_prover(
     merlin
         .fill_challenge_scalars(&mut r)
         .expect("Failed to extract challenge scalars from Merlin");
-
     // let a = sum_fhat_1, b = sum_fhat_2, c = sum_fhat_3 for brevity
     let ((mut a, mut b, mut c), mut eq) = rayon::join(
         || calculate_witness_bounds(r1cs, z),
         || calculate_evaluations_over_boolean_hypercube_for_eq(r),
     );
-
+    pad_min2_pow2(&mut a);
+    pad_min2_pow2(&mut b);
+    pad_min2_pow2(&mut c);
+    pad_min2_pow2(&mut eq);
     let mut alpha = Vec::<FieldElement>::with_capacity(m_0);
 
     let blinding_polynomial = generate_blinding_spartan_univariate_polys(m_0);
 
-    let blinding_polynomial_for_commiting = EvaluationsList::new(pad_to_power_of_two(
-        blinding_polynomial.iter().flatten().cloned().collect(),
-    ));
+    let nv_b = whir_for_blinding_of_spartan_config
+        .mv_parameters
+        .num_variables;
+    let target_b = 1usize << (nv_b - 1);
+
+    // Flatten and pad to exactly 1 << nv_b
```

Comment:
The comment says `"pad to exactly 1 << nv_b"` but the code checks i`f flat.len() < target_b `where `target_b = 1 << (nv_b - 1)`. This is inconsistent.